### PR TITLE
Define path with __dir__

### PR DESCRIPTION
**Pull request by bogdanvlviv**
> "... with \_\_dir\_\_ we can restore order in the Universe." - by @fxn.

This words inspired me to restore order in Rails and my own projects.

I read [Contributing to Ruby on Rails](http://guides.rubyonrails.org/contributing_to_ruby_on_rails.html) and know about "cosmetic changes", but in this case, i think it something more - Very often new code written under the influence of old code so this PR will inspire to define a path with `__dir__` in this case.
We have some examples of code, that includes "defining a path", in our guide -  it also important.

Also, with `__dir__`, code in some cases became more readable. Example:
```diff
- append_view_path File.expand_path(File.join(File.dirname(__FILE__), "views"))
+ append_view_path File.expand_path("views", __dir__)
```

Feel free to close this PR if something wrong.

Related to 5b8738c2df003a96f0e490c43559747618d10f5f
r? @fxn

### CODE REVIEW

**Comment by fxn** [comment]
Looking good to me, I'll do a careful revision of the patch later.

File: activejob/Rakefile [hunk 1]
```diff
@@ -44,9 +44,8 @@ namespace :test do
 
     namespace :isolated do
       task adapter => "test:env:#{adapter}" do
-        dir = File.dirname(__FILE__)
-        Dir.glob("#{dir}/test/cases/**/*_test.rb").all? do |file|
-          sh(Gem.ruby, "-w", "-I#{dir}/lib", "-I#{dir}/test", file)
+        Dir.glob("#{__dir__}/test/cases/**/*_test.rb").all? do |file|
+          sh(Gem.ruby, "-w", "-I#{__dir__}/lib", "-I#{dir}/test", file)
```
Comment (matthewd) [inline activejob/Rakefile]: :bug:

**Review by matthewd — CHANGES_REQUESTED** [review]
I only saw one error. But let's squash down to a single commit; separating by component doesn't seem helpful to history.

File: activejob/Rakefile (hunk 1 above)
Comment (bogdanvlviv) [inline activejob/Rakefile]: :+1:

**Review by jeremy — APPROVED** [review]
Nice. Some of the switches to use expand_path may be unnecessary. That expands symlinks and such, which may not be desirable. All the cases I checked looked good, though.

On that tack, how many of these File.expand_path do we really need? It seems to be more a style convention than a functional concern in many cases.

(We have a good number of `require globbed_absolute_path` that we can convert to `require_relative`, too!)

**Comment by fxn** [comment]
I have made a first pass.

As you surely know, `File.dirname(__FILE__)` returns a path relative to the current working directory, whereas `__dir__` yields an absolute path. In that sense, there are a bunch of edits that are not equivalent because they replace one with the other, for example

```ruby
local_gemfile = File.dirname(__FILE__) + "/.Gemfile"
```

and

```ruby
local_gemfile = File.expand_path(".Gemfile", __dir__)
```

are different. Some of these end up in constants, which are public, but my impression is that the replacements make sense, and indeed in some spots the relative path was brittle in comparison.

So, in principle I would say we are good there.

There are a handful of unnecessary trailing slashes, for example:

```ruby
File.expand_path("../", __dir__)
```

`File.expand_path` does not need that slash, and it is going to be removed from the return value anyway. Could you please clean them all? Grepping for `/"` would show them.

On the other hand

```ruby
[File.expand_path(__dir__)]
```

performs an unnecessary call to `File.expand_path` because as I said above `__dir__` is already absolute.

That was a great patch, let's refine a bit!

**Review by fxn — CHANGES_REQUESTED** [review]
Some changes were requested in a comment.

**Comment by bogdanvlviv** [comment]
will do.
@fxn  I am not sure about repace `File.expand_path(__dir__) ` to `__dir__`
```ruby
#  absolute path
File.expand_path(__dir__) # => "/home/username/rails/"
# canonicalized absolute path
__dir__ # => "."
```

**Comment by fxn** [comment]
[`__dir__` returns an absolute path](https://ruby-doc.org/core-2.4.1/Kernel.html#method-i-__dir__), except in IRB, where `__FILE__` has also a special value.

**Comment by bogdanvlviv** [comment]
i need a sleep, i will ready to finish this patch tomorrow.

**Comment by fxn** [comment]
Rest @bogdanvlviv :), we also need our own time to review. The code has been that way for a long time, a few days more won't make a difference 😄 . Take your time!

**Comment by bogdanvlviv** [comment]
@fxn I just passed through the PR one more time. Need your new review.

**Comment by bogdanvlviv** [comment]
> (We have a good number of require globbed_absolute_path that we can convert to require_relative, too!)

@jeremy i like this idea.
i am ready to open the new patch "`require_relative` instead of `require` with absolute path".
Can you confirm that rails ready to go in this way?

**Review by fxn — APPROVED** [review]

**Pull request merged**